api-reference/10 UI Components/dxScheduler/1 Configuration/appointmentTooltipTemplate.md

```diff
@@ -29,6 +29,374 @@ A template name or container.
+            this.scheduler.instance.hideAppointmentTooltip();
+        }
+    
+        isDeleteButtonExist({ appointmentData }: { appointmentData: dxSchedulerAppointment}): boolean {
+            const schedulerInstance = this.scheduler.instance;
+            const isAppointmentDisabled = appointmentData.disabled;
+            const isDeleteAllowed = (schedulerInstance.option('editing') && schedulerInstance.option('editing.allowDeleting') === true)
+            || schedulerInstance.option('editing') === true;
+            
+            return !isAppointmentDisabled && isDeleteAllowed;
+        }
+    }
+
+    <!-- tab: tooltip.component.html -->
+    <div class="dx-tooltip-appointment-item">
+        <div class="dx-tooltip-appointment-item-marker">
+            <div
+                class="dx-tooltip-appointment-item-marker-body"
+                [style.background-color]="markerColor" 
+            ></div>
+        </div>
+
+        <div class="dx-tooltip-appointment-item-content">
+            <div class="dx-tooltip-appointment-item-content">
+                <div class="dx-tooltip-appointment-item-content-subject">
+                    {{appointmentData.text}}
+                </div>
+                <div class="dx-tooltip-appointment-item-content-date">
+                    {{appointmentData.startDate}}
+                </div>
+            </div>
+        </div>
+        <div
+            *ngIf="isDeleteButtonExist"
+            class="dx-tooltip-appointment-item-delete-button-container"
+        >
+            <dx-button
+                class="dx-tooltip-appointment-item-delete-button"
+                icon="trash"
+                stylingMode="text"
+                (onClick)="onDeleteButtonClick.emit($event)"
+            ></dx-button>
+        </div>
+    </div>
+
+    <!-- tab: tooltip.components.ts -->
+    @Component({
+        selector: "Tooltip",
+        templateUrl: "./tooltip.component.html"
+    })
+    export class TooltipComponent {
+        @Input() appointmentData: dxSchedulerAppointment;
+        @Input() markerColor: string;
+        @Input() isDeleteButtonExist: boolean;
+        @Output() onDeleteButtonClick = new EventEmitter<ClickEvent>();
+    }
+
+    @NgModule({
+        imports: [DxButtonModule, CommonModule],
+        declarations: [TooltipComponent],
+        exports: [TooltipComponent]
+    })
+
+    export class TooltipModule {}
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxScheduler
+            :ref="schedulerRefKey"
+            ...
+            appointment-tooltip-template="appointmentTooltipTemplate"
+        >
+            <template #appointmentTooltipTemplate="{ data }">
+                <Tooltip
+                    :data="data"
+                    @delete-button-click="onDeleteButtonClick($event, data)"
+                    :markerColor="getColor(data.appointmentData.employeeID)"
+                    :isDeleteButtonExist="getDeleteButtonStatus(data)"
+                />
```

Comment:
Let's change these lines:
- Make the event the last property
- Change color to the constant value
- Add ```NOTE``` comment
- Change the ```getDeleteButtonStatus``` -> ```isDeleteButtonExist``` (to be consistent with other platforms)

```suggestion
      <!-- NOTE: You can get color from resources with data.appointmentData.resouceId -->
      <Tooltip
        :data="data"
        :markerColor="'#337ab7'"
        :isDeleteButtonExist="isDeleteButtonExist(data)"
        @delete-button-click="onDeleteButtonClick($event, data)"
      />
```